types/validator_set.go

```diff
@@ -63,7 +68,13 @@ func (valSet *ValidatorSet) IncrementAccum(times int) {
 		if i == times-1 {
 			valSet.Proposer = mostest
 		}
-		mostest.Accum -= int64(valSet.TotalVotingPower())
+
+		// mind underflow
```

Comment:
I'd add underflow protection to addSafe() as well.  maybe we can have subSafe() too, which handles both over/underflow protection.